monitors/BalanceMonitor.js

```diff
@@ -0,0 +1,138 @@
+const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
+
+class BalanceMonitor {
+  constructor(logger, tokenBalanceClient, account, botsToMonitor) {
+    this.logger = logger;
+    this.account = account;
+
+    // An array of bot objects to monitor. Each bot's `botName` `address`,
+    // `CollateralThreshold` and`syntheticThreshold` must be given. Example:
+    // [{ name: "Liquidator Bot",
+    //   address: "0x12345"
+    //   collateralThreshold: x1,
+    //   syntheticThreshold: x2,
+    //   etherThreshold: x3 },
+    // ...]
+    this.botsToMonitor = botsToMonitor;
+
+    // Structure to monitor if a wallet address have been alerted yet for each alert type.
+    this.walletsAlerted = {};
+
+    // Populate walletsAlerted for each bot, starting with each alert type at not sent.
+    for (let bot of botsToMonitor) {
+      this.walletsAlerted[bot.address] = {
```

Comment:
Can you explain how this works? Let's assume the balance is random on every query. In what cases would I get an alert and in what cases would I not?

Reply:
this works by iterating over all the bots specified in the `botsToMonitor` constructor param and for each bot checking if their balance of a given asset is below the provided threshold. I have added a comment to explain this a bit better.